type_templates/vrm.js

```diff
@@ -77,12 +77,99 @@ const _addAnisotropy = (o, anisotropyLevel) => {
   });
 };
 
+const _setQuality = async (quality, app) => {
+  const skinnedVrms = app.skinnedVrms;
+  const baseVrm = skinnedVrms.base;
+
+  const _swapMaterials = async (type) => {
+
+    //actually do the swap
+    switch (type) {
+      case "toon": {
+
+        let update = Object.keys(app.materials.toon).length > 0;
+
+        skinnedVrms.base.scene.traverse((object) => {
+          if (object.material && app.isBasic(object.material)) {
+            const name = object.material.name;
+            app.setMaterial(name, 'base', object.material);
+            update && (object.material = [app.materials.toon[name]]);
+          }
+
+        });
+        break;
+
+      }
+      default: {
+        let update = false;
+        const target = skinnedVrms.base;
+        target.scene.traverse((object) => {
+          if (!update && object.material && app.isToon(object.material)) {
+            update = true;
+          }
+        });
+
+        target.scene.traverse((object) => {
+          if (object.material && app.isToon(object.material)) {
+            const name = object.material[0].name;
+            update && app.setMaterial(name, 'toon', object.material[0]);
+
+            object.material = app.materials.base[name];
+          }
+        });
+      }
+    }
+  }
+
+  switch (quality ?? 'MEDIUM') {
+    case 'LOW':
+    case 'MEDIUM':
+    case 'HIGH': {
+      await _swapMaterials();
+      baseVrm.scene.name = "base mesh"
+      app.setActive('base');
+      break;
+    }
+    case 'ULTRA': {
+      if (skinnedVrms.toon) {
+
+      } else {
+        if (!skinnedVrms.toon) {
+
+          await _toonShaderify(baseVrm);
+          skinnedVrms['toon'] = baseVrm;
+          skinnedVrms.toon.scene.name = 'base-tooned';
+        }
+      }
+
+      await _swapMaterials("toon");
+      app.setActive('toon');
+      break;
+    }
+    default: {
+      throw new Error('unknown avatar quality: ' + quality);
+    }
+  }
+  return app.getActive();
+}
 
 export default e => {
   const physics = usePhysics();
 
   const app = useApp();
   app.appType = 'vrm';
+  app.active = 'base';
+
+  //make sure we have materials to work with
+  app.materials = {
+    toon: {},
+    base: {}
+  };
+  app.isToon = material => material[0] && material[0].isMToonMaterial;
+  app.isBasic = material => material.type == "MeshBasicMaterial" && material.name; //we're only changing named materials

```

Comment:
Try not to base anything on THREE.js `name`, that is mostly for user convenience.